### ISSUE TITLE
[Search Enging]:K1M: CA1 Radial Trilaminar to CA1 Pyramidal connection

### ISSUE THREAD

**Issue by k1moradi** [Hippocampome-Org/php#369]
The following search query does not produce any result without producing errors while it should:

Connection:(Presynaptic:(Neurotransmitter:Inhibitory AND Morphology:(Dendrites:CA1:?222 AND Axons:CA1:01?1) AND Markers:(D+:CCK OR I+:CCK)), Postsynaptic:(Morphology:(Dendrites:CA1:2222 AND Axons:CA1:0001 AND Soma:CA1:??1?) AND Name:Pyramidal))

sUIDs  | Presynaptic Cell Types | dUIDs | Postsynaptic Cell Types
4013 | CA1 Radial Trilaminar (-)2333 | 4000 | CA1 Pyramidal (+)2223p
4080 | CA1 Bistratified (-)0333 | 4000 | CA1 Pyramidal (+)2223p

Removing marker section Markers:(D+:CCK OR I+:CCK) it will show results 

Connection:(Presynaptic:(Neurotransmitter:Inhibitory AND Morphology:(Dendrites:CA1:?222 AND Axons:CA1:01?1)), Postsynaptic:(Morphology:(Dendrites:CA1:2222 AND Axons:CA1:0001 AND Soma:CA1:??1?) AND Name:Pyramidal))

sUIDs  | Presynaptic Cell Types | dUIDs | Postsynaptic Cell Types
-- | -- | -- | --
4013 | CA1 Radial Trilaminar (-)2333 | 4000 | CA1 Pyramidal (+)2223p
4080 | CA1 Bistratified (-)0333 | 4000 | CA1 Pyramidal (+)2223p
4081 | CA1 Ivy (-)0333 | 4000 | CA1 Pyramidal (+)2223p

.

**Comment by drdiek** [comment]
Nikhil,

I believe you showed me a solution to this on your laptop. Has the modified code made its way to csv2db yet? If so, how do we test it?

Sincerely,

Diek

**Comment by k1moradi** [comment]
Yes, I asked for the functionality. Thanks for the implementation. I will update my queries.

**Comment by k1moradi** [comment]
So should I use "+-" or "-+" or "±" or "∓"? or they are interchangeable?

**Comment by drdiek** [comment]
Keivan,

I believe you can only use "+-."

Sincerely,

Diek

**Comment by k1moradi** [comment]
(1) Then this issue should remain open since users should get a meaningful error if the search engine does not recognize the provided syntax. For instance, an appropriate error here might be 

**"Syntax Error: After name "Markers", 
":" or ":(" is expected, then 
"D" or "I", then 
"+:", "-:", or "+-.:", then 
one of hippocampom.org markers**

(2) The "+-." syntax is also nonintuitive.  Is there any good reason for using this syntax?
I think users should type "+-", "-+", or "±" which should automatically be converted to "±" on the user side before sending it to the server side to get the results.

**Comment by drdiek** [comment]
Since the original Issue has been addressed, I am opening a new ticket to address Keivan's newest concerns:

KM: [markers] The connectivity search engine "+-" syntax is nonintuitive #388
